### PR TITLE
CE-1283 Add new 'poweruser' group for better tracking

### PR DESCRIPTION
**Pull request by adamkarminski**
[See CE-1283](https://wikia-inc.atlassian.net/browse/CE-1283)

This pull request binds a new 'poweruser' group with being a Power User. Using _user_groups_ to store information about Power Users allows us to use currently gathered statistics in our favor (most of them include users' groups).

It also implements a new method to get current PowerUser-related properties for a user.

Ping @Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/PowerUser/PowerUser.class.php [hunk 1]
```diff
@@ -69,44 +86,141 @@ function __construct( \User $oUser ) {
...
+	 * Adds group rights to a user if the property's name
+	 * matches one in the aPowerUsersRightsMapping array
+	 *
+	 * @param string $sProperty One of the types in consts
+	 * @return bool
+	 */
+	public function addPowerUserAddGroup( $sProperty ) {
+		if ( in_array( $sProperty, self::$aPowerUsersRightsMapping )
+			&& !in_array( self::GROUP_NAME, $this->oUser->getGroups() ) )
+		{
```
Comment (lukaszkonieczny) [inline extensions/wikia/PowerUser/PowerUser.class.php]: Closing bracket ```)``` should be before ```{``` (http://www.mediawiki.org/wiki/Manual:Coding_conventions#Line_continuation)

File: extensions/wikia/PowerUser/PowerUser.class.php [hunk 2]
```diff
@@ -69,44 +86,141 @@ function __construct( \User $oUser ) {
...
+		return $aUserTypes;
+	}
+
+	/**
+	 * Perform all actions to make a user a PowerUser
 	 *
 	 * @param string $sProperty One of the types in consts
 	 * @return bool
 	 */
 	public function addPowerUserProperty( $sProperty ) {
```
Comment (lukaszkonieczny) [inline extensions/wikia/PowerUser/PowerUser.class.php]: This function will return false if user is already in group

Comment (adamkarminski) [inline extensions/wikia/PowerUser/PowerUser.class.php]: Fixed!

**Comment by lukaszkonieczny** [comment]
lgtm

**Pull request merged**